MarkBind/markbind
# Update logger to work with progress bar

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #1723, which is related to the first task of #1633

- ~~Some patching is done to the node-progress package to add helper methods that allow loggers to interrupt the progress bar~~
- ~~Update the logger packages (`winston` and `winston-daily-rotate-file`)~~
	- ~~some patching is done to ensure that error stack trace is properly printed out~~
- ~~Adjustments were made to preserve the CLI output, with the exception that the new version of winston prepends 3 white space characters before the message, hence the gap between the level label and the message is slightly wider:~~
  - ~~before~~
  - ![image](https://user-images.githubusercontent.com/41845017/153006359-00c50285-60da-4cbb-a153-8206b95ae74e.png)
  - ~~after~~
  - ![image](https://user-images.githubusercontent.com/41845017/163705516-c54e1289-98c3-492c-8a2a-b06a523478b4.png)

**Anything you'd like to highlight / discuss:**

~~Irrelevant observation: In the core package's `package-lock.json`, the version of core is still 3.0.6. Actually not sure if this lock file needs to exist at all...~~
(resolved)

**Testing instructions:**
- checkout the PR
- npm run setup to update the dependencies
- run `markbind serve`/`markbind build` to verify the CLI output
- also check the `_markbind/logs` to see the content of .log files

Can consider running `npm run test`, which will have quite a bit of console logging that is resulted from running all the functional tests, hence can observe the console behavior.



**Proposed commit message: (wrap lines at 72 characters)**
Update logger to work with progress bar

When the page build task is ongoing, the progress bar
shown is interrupted unexpectedly by warnings or error
logs, which will be written to the console without any
preceding linebreak.

Let's patch the node-progress package to signal the
start and the end of a proper interruption, and have
the loggers invoke these methods before and after
logging. Let's also upgrade the logger dependencies.

When the progress bar is running, the loggers will
call the relevant method to clear off the current bar
before logging and redraws the progress bar afterward.



---

**Checklist:** :ballot_box_with_check:
